lib/primitives/address.js

```diff
@@ -320,7 +325,9 @@ class Address extends bio.Struct {
    */
 
   fromProgram(version, hash) {
-    assert(version >= 0, 'Bad version for witness program.');
+    assert(version >= consensus.MIN_WITNESS_PROGRAM_VERSION
```

Comment:
Interesting that a max version wasn't enforced here already - do you know why? Might need to add an extra check to bcoin...

Reply:
`fromProgram` is not part of the consensus code path in either `hsd` or `bcoin`